src/lib.rs

```diff
@@ -747,6 +747,11 @@ pub struct IterMut<'a, K: 'a, V: 'a> {
     marker: marker::PhantomData<(&'a K, &'a mut V)>,
 }
 
+/// A consuming insertion-order iterator over a `LinkedHashMap`'s entries.
+pub struct IntoIter<K, V, S> {
```

Comment:
If we can't mention `S`, we can't keep the LinkedHashMap. How about collecting the values into a vector and iterating that?

HashMap gets around this by keeping the private RawTable that backs the HashMap.